### ISSUE TITLE
Batch size of 2 will break the training loop on LTX-loRA finetuning.

### ISSUE THREAD

**Issue by ArEnSc** [a-r-r-o-w/finetrainers#173]
### System Info / 系統信息

RuntimeError('The size of tensor a (2) must match the size of tensor b (128) at non-singleton dimension 2')

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

Set batch size to 2
errors out at runtime on this
 noisy_latents = (1.0 - sigmas) * latent_conditions["latents"] + sigmas * noise

### Expected behavior / 期待表现

To support batch sizes larger than 2

**Comment by sayakpaul** [comment]
This issue was also encountered in https://github.com/a-r-r-o-w/finetrainers/pull/139

**Comment by neph1** [comment]
Also here: https://github.com/a-r-r-o-w/finetrainers/issues/132
But I did a run the other day with batch size 4, so it seems it's dependent on other factors than just batch size.

**Comment by sayakpaul** [comment]
You mean the issue is resolved?

**Comment by neph1** [comment]
It worked for me on the 31st. But since this issue is newer, there might be other factors involved.

**Comment by sayakpaul** [comment]
Could you try it again?

**Comment by neph1** [comment]
Not for a couple of days.

**Comment by ArEnSc** [comment]
Ill get you guys more info, the LTX paper just dropped so grokking that rn

**Comment by neph1** [comment]
Just pulled the latest and tried with the same params as last time:

01/05/2025 20:28:42 - INFO - finetrainers - Training configuration: {
    "trainable parameters": 234881024,
    "total samples": 16,
    "train epochs": 3000,
    "train steps": 3000,
    "batches per device": 4,
    "total batches observed per epoch": 4,
    "train batch size": 16,
    "gradient accumulation steps": 4
}

Training starts without any problems.

I might test with some other params tomorrow.

**Comment by a-r-r-o-w** [comment]
Can't seem to replicate either... I know we don't support batch_size > 1 when precomputation is enabled (because there is no BucketSampler for latent tensors), but normal training without precomputation seems to work as expected for batch sizes 1, 2, 4

**Comment by neph1** [comment]
Some more variations tested:

> 01/06/2025 07:32:13 - INFO - finetrainers - Training configuration: {
>     "trainable parameters": 234881024,
>     "total samples": 16,
>     "train epochs": 750,
>     "train steps": 3000,
>     "batches per device": 4,
>     "total batches observed per epoch": 4,
>     "train batch size": 4,
>     "gradient accumulation steps": 1
> }

01/06/2025 13:06:28 - INFO - finetrainers - Training configuration: {
    "trainable parameters": 234881024,
    "total samples": 16,
    "train epochs": 375,
    "train steps": 3000,
    "batches per device": 2,
    "total batches observed per epoch": 8,
    "train batch size": 2,
    "gradient accumulation steps": 1
}

**Comment by ArEnSc** [comment]
ill give it a shot again today thanks for trying everyone

**Comment by a-r-r-o-w** [comment]
@ArEnSc LMK if this is still on issue or we could mark this as closed

**Comment by ArEnSc** [comment]
> @ArEnSc LMK if this is still on issue or we could mark this as closed

ill try this tonight

**Comment by a-r-r-o-w** [comment]
Arbitrary batch size is now supported for all models out-of-the-box. There are some exceptions like CogView4 (which uses custom sequence length for each text embedding leading to concatenation of tensors not being possible but I'm looking into the NestedTensor's API to support this), which will be improved gradually.

There are also unit tests to check combinations of different parallelism degrees with different batch size, and a list of fully-reproducible examples in `examples/`. I look forward to this being very stable for the foreseeable future, so please LMK if you notice any problems
